hyperjumptech/monika
# Adding JSON Schema for Monika configuration

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Implements issue #723 

## How did you implement / how did you fix it  
1.  Add new file `monika-configuration.json` to be used for YAML validation.

## How to test  
0. Easiest is to chuck the json schema to an online validator, and put in monika.jsons to validate (see note below)
1.  To test on your Visual Code you will need to install an extension for [YAML language support by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml)
2. Once installed you need to get the extension to point to our `monika-configuration.json`
3. Goto File -> Preferences -> Settings in Visual Code and edit the settings.json

![image](https://user-images.githubusercontent.com/964106/177102479-bed5a9bf-b384-403e-b1f1-1bf449189b3b.png)

4. Set the entry to point to `monika-configuration.yml` and map it to `monika.yml`

![image](https://user-images.githubusercontent.com/964106/177102618-704189d0-5076-4037-b08a-778675cf8e31.png)

## Screenshots

1. Description and error validation

![image](https://user-images.githubusercontent.com/964106/177104370-c5aeb437-bdaa-4843-ad38-b282fe12f4b0.png)


note: some field names are specific for Schema Store ("fileMatch", "url" and "name") and may cause errors when using strict json validators (such as ajv)

Co-Authored-by: @nicnocquee
